src/org/zaproxy/zap/extension/ascanrulesBeta/HPP.java

```diff
@@ -154,7 +157,12 @@ public void scan() {
 					log.debug("Injecting payload...");
 					HttpMessage newMsg = getNewMsg();
 					newMsg.setGetParams(tags);
-					sendAndReceive(newMsg);
+					try {
+						sendAndReceive(newMsg);
+					} catch (IllegalStateException|UnknownHostException ex) {
+						log.info("Caught " + ex.getClass().getName() + " when accessing: " + newMsg.getRequestHeader().getURI().toString() + 
+								"\n The target may have replied with a poorly formed redirect due to our input.");
```

Comment:
Do you really think so? In the example above getMessage() just returns:
> 92116 [ZAP-ActiveScanner-1] INFO org.zaproxy.zap.extension.ascanrulesBeta.HPP  - &zap=zaproxy

If you really think that adds value I can put it in, but to me it isn't terribly helpful.